helix-rest/src/main/java/org/apache/helix/rest/server/resources/zookeeper/ZooKeeperAccessor.java

```diff
@@ -0,0 +1,181 @@
+        return getData(path, cmd);
+      case getStringData:
+        return getData(path, cmd);
+      case getChildren:
+        return getChildren(path);
+      default:
+        LOG.error("Unsupported command :" + commandStr);
+        return badRequest("Unsupported command :" + commandStr);
+    }
+  }
+
+  /**
+   * Checks if a ZNode exists in the given path.
+   * @param path
+   * @return true if a ZNode exists, false otherwise
+   */
+  private Response exists(String path) {
+    if (!isPathValid(path)) {
+      String errMsg = "exists(): The given path {} is not a valid ZooKeeper path!" + path;
+      LOG.error(errMsg);
+      return badRequest(errMsg);
+    }
+
+    Map<String, Boolean> result = ImmutableMap.of(ZooKeeperCommand.exists.name(),
+        _zkBaseDataAccessor.exists(path, AccessOption.PERSISTENT));
+    return JSONRepresentation(result);
+  }
+
+  /**
+   * Reads the given path from ZooKeeper and returns the binary data for the ZNode.
+   * @param path
+   * @param command denotes whether return type should be binary or String
+   * @return binary data in the ZNode
+   */
+  private Response getData(String path, ZooKeeperCommand command) {
+    if (!isPathValid(path)) {
+      String errMsg = "getData(): The given path {} is not a valid ZooKeeper path!" + path;
+      LOG.error(errMsg);
+      return badRequest(errMsg);
+    }
+
+    if (_zkBaseDataAccessor.exists(path, AccessOption.PERSISTENT)) {
+      byte[] bytes = _zkBaseDataAccessor.get(path, null, AccessOption.PERSISTENT);
+      String s = new String(bytes);
+
+      switch (command) {
+        case getBinaryData:
+          Map<String, byte[]> binaryResult =
+              ImmutableMap.of(ZooKeeperCommand.getBinaryData.name(), bytes);
+          // Note: this serialization (using ObjectMapper) will convert this byte[] into
+          // a Base64 String! The REST client (user) must convert the resulting String back into
+          // a byte[] using Base64.
+          return JSONRepresentation(binaryResult);
+        case getStringData:
+          Map<String, String> stringResult =
+              ImmutableMap.of(ZooKeeperCommand.getStringData.name(), new String(bytes));
+          return JSONRepresentation(stringResult);
+        default:
+          LOG.error("Unsupported command :" + command);
+          return badRequest("Unsupported command :" + command);
+      }
+    } else {
+      throw new WebApplicationException(Response.status(Response.Status.NOT_FOUND)
+          .entity(String.format("The ZNode at path %s does not exist", path)).build());
+    }
+  }
+
+  /**
+   * Returns a list of children ZNode names given the path for the parent ZNode.
+   * @param path
+   * @return list of child ZNodes
+   */
+  private Response getChildren(String path) {
+    if (!isPathValid(path)) {
+      String errMsg = "getChildren(): The given path {} is not a valid ZooKeeper path!" + path;
+      LOG.error(errMsg);
+      return badRequest(errMsg);
+    }
```

Comment:
1. {} doesn't apply here.
2. This is a code block that could be reused in other methods. Can we make it a method to reduce duplicate code?
3. Shall we really need the log level at `error`? Maybe other reviewers could give opinions. @lei-xia @jiajunwang @i3wangyi

Reply:
1. Fixed. Good catch!
2. This code needs some customization to differentiate the log message. If you have a better idea on how to refactor, please give a suggestion.
3. This should be an info call as mentioned above since it doesn't indicate there was an error on the server logic.

Reply:
2. I think for the response, the method name is not necessary. I would not expect the java method name in a Rest API response because my command is already provided in the API. Instead, the message "The given path is not a valid ZooKeeper path: " is good enough. 
In the server log, I agree it is OK to add the info for clearly debugging. And maybe a more generic message "Getting children" is better than the method name itself?

4. One more thing, from the user perspective, I am not sure if we would like provide a specific message for the case with trailing slash: `/abc/`, which I think it might be easy to make a mistake by adding the trailing slash.
Shall we provide a message for in the bad request specifically to help users realize the error? If a user thinks a path with trailing slash is OK, it would be confusing and time costly for the user to troubleshoot.

Reply:
2. I moved the common path-checking logic to the get() function.
3. This sounds like a nice feature, but we could add this in the future. The Javadoc contains this and the user manual will also contain this information, so I don't think it's completely necessary right now.